src/Functions/DateTimeTransforms.h

```diff
@@ -57,25 +60,51 @@ struct ZeroTransform
     static UInt16 execute(UInt16, const DateLUTImpl &) { return 0; }
 };
 
+template <DateTimeOverflowMode date_time_overflow_mode = default_date_time_overflow_mode>
 struct ToDateImpl
 {
     static constexpr auto name = "toDate";
 
     static UInt16 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
     {
-        return static_cast<UInt16>(time_zone.toDayNum(t.whole));
+        return execute(t.whole, time_zone);
```

Comment:
Sure `t.whole` is always `UInt16`? Asking because it is part of `DecimalComponents` and dependent on a templated type (`using T = typename DecimalType::NativeType;`). There is a risk, that we call an unwanted overload here.

Reply:
> Sure t.whole is always UInt16

No, here it is always Int64